### PR TITLE
Fetch Jobs using GraphQL in portal

### PR DESCRIPTION
**Pull request by laurianed**
Due to GraphQL fetching mode (using cursors instead of pages) Tasks and Jobs are no longer displayed the same way so views and models are now different for Jobs and Tasks

### CODE REVIEW

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/Job.java [hunk 1]
```diff
@@ -344,12 +344,12 @@ public boolean isExecuted() {
      * @return a POJO equivalent
      */
     public static Job parseJson(JSONObject jsonJob) {
-        JSONObject jsonInfo = jsonJob.get("jobInfo").isObject(); // TODO to update jobInfo
+        JSONObject jsonInfo = jsonJob.get("node").isObject(); // TODO to update jobInfo
```
Comment (tobwiens) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/Job.java]: can you remove the TODO comment

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/Job.java [hunk 2]
```diff
@@ -392,12 +392,15 @@ public static Job parseJSONInfo(JSONObject jsonJobInfo) {
 
     private static Map<String, String> extractMap(JSONValue mapValue) {
         if (mapValue != null) {
-            JSONObject mapObject = mapValue.isObject();
-            if (mapObject != null) {
-                Set<String> keySet = mapObject.keySet();
-                Map<String, String> resultMap = new HashMap<>(keySet.size());
-                for (String key : keySet) {
-                    resultMap.put(key, mapObject.get(key).isString().stringValue());
+            JSONArray keyValueArray = mapValue.isArray();
```
Comment (tobwiens) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/Job.java]: Is this a standard problem which you could use a library for?

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/controller/JobsController.java [hunk 3]
```diff
@@ -356,20 +361,27 @@ public void fetchJobs(boolean showUpdating) {
...
-        int limit = paginationController.getModel().getPageSize();
+        String startCursor = paginationController.getModel().getStartCursor();
+        String endCursor = paginationController.getModel().getEndCursor();
+        int pageSize = SchedulerConfig.get().getPageSize(PaginatedItemType.JOB);
+        boolean first = paginationController.getModel().isFirst();
 
         ExecutionsModel executionModel = this.parentController.getModel();
-        boolean fetchMyJobs = executionModel.isFetchMyExecutionsOnly();
+        String user = null;
+        if (executionModel.isFetchMyExecutionsOnly())
```
Comment (tobwiens) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/controller/JobsController.java]: can you put the { } around the if

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/controller/TasksPaginationController.java [hunk 4]
```diff
@@ -57,4 +58,104 @@ public Layout buildView() {
...
+
+    /**
+     * Gets the text that displays the pagination status.
+     * @return the text that displays the pagination status.
+     */
+    public String getPaginationRangeLabel() {
+        int page = this.model.getPage();
+        int size = this.model.getPageSize();
+
+        long index = page * size;
```
Comment (tobwiens) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/controller/TasksPaginationController.java]: Can you compose this method into smaller methods where each method name explains what it does

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/model/JobsPaginationModel.java [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+
+import org.ow2.proactive_grid_cloud_portal.scheduler.client.SchedulerListeners.PaginationListener;
+
+
+/**
+ * A model for the pagination of items.
+ * @author activeeon team
+ *
+ */
+public class JobsPaginationModel extends PaginationModel {
```
Comment (tobwiens) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/model/JobsPaginationModel.java]: Is there a way to the XPaginationModel Classes immutable?

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/Job.java (hunk 2 above)
Comment (laurianed) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/Job.java]: Yes but it needs to be GWT client-side compatible and jackson is not

File: scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/model/JobsPaginationModel.java (hunk 5 above)
Comment (laurianed) [inline scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/model/JobsPaginationModel.java]: but we don't want them to be

**Pull request merged**